activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -520,7 +520,15 @@ def collation
         show_variable 'collation_database'
       end
 
-      def tables(name = nil, database = nil, like = nil) #:nodoc:
+      def tables #:nodoc:
+        sql = "SHOW FULL TABLES WHERE table_type = 'BASE TABLE'"
+
+        execute_and_free(sql, 'SCHEMA') do |result|
+          result.collect(&:first)
+        end
+      end
+
+      def tables_and_views(name = nil, database = nil, like = nil) #:nodoc:
```

Comment:
Also can we make this method private? It is not being used anywhere besides test.